### PR TITLE
Python package updates / addition of lz4 

### PR DESCRIPTION
**Pull request by davidlange6**
adding lz4 instead of picking up from inside of root
adding python interfaces to lz4 and lzma
updating to latest version of a number of python packages

### CODE REVIEW

**Comment by davidlange6** [comment]
please test




On 9 Jun 2018, at 11:30, David Lange <notifications@github.com<mailto:notifications@github.com>> wrote:


adding lz4 instead of picking up from inside of root
adding python interfaces to lz4 and lzma
updating to latest version of a number of python packages

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/cms-sw/cmsdist/pull/4103

Commit Summary

  *   python version updates, add lzma interface
  *   add lz4
  *   add lz4
  *   add lzma python interface
  *   fix spec

File Changes

  *   A lz4-toolfile.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-0> (22)
  *   A lz4.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-1> (19)
  *   M py2-Theano.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-2> (2)
  *   A py2-backports_lzma.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-3> (12)
  *   M py2-bleach.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-4> (2)
  *   M py2-certifi.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-5> (2)
  *   M py2-cython.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-6> (2)
  *   M py2-flawfinder.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-7> (2)
  *   M py2-fs.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-8> (2)
  *   M py2-h5py.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-9> (2)
  *   M py2-jupyter_client.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-10> (2)
  *   M py2-llvmlite.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-11> (2)
  *   A py2-lz4.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-12> (11)
  *   M py2-oamap.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-13> (2)
  *   M py2-packaging.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-14> (2)
  *   M py2-pandas.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-15> (2)
  *   M py2-pexpect.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-16> (2)
  *   M py2-psutil.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-17> (2)
  *   M py2-pyopenssl.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-18> (2)
  *   M py2-python-dateutil.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-19> (2)
  *   M py2-pytz.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-20> (2)
  *   M py2-scikit-learn.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-21> (2)
  *   M py2-tqdm.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-22> (2)
  *   M py2-uproot.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-23> (2)
  *   M py2-xgboost.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-24> (16)
  *   M python_tools.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-25> (1)
  *   M root.spec<https://github.com/cms-sw/cmsdist/pull/4103/files#diff-26> (6)

Patch Links:

  *   https://github.com/cms-sw/cmsdist/pull/4103.patch
  *   https://github.com/cms-sw/cmsdist/pull/4103.diff

File: lz4-toolfile.spec [hunk 1]
```diff
@@ -0,0 +1,22 @@
+### RPM external lz4-toolfile 1.0
+Requires: lz4
+
+%prep
+%build
+%install
+mkdir -p %i/etc/scram.d
+cat << \EOF_TOOLFILE > %i/etc/scram.d/xz.xml
```
Comment (smuzaffar) [inline lz4-toolfile.spec]: @davidlange6 , why tool name is xz and not lz4?

**Comment by davidlange6** [comment]
hi @smuzaffar - thanks for noticing - its a mistake

**Comment by davidlange6** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
@davidlange6 , can you please check the comparison results?

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
@davidlange6 , can you please check the comparison for lz4 again. It again shows 25 errors.

**Comment by davidlange6** [comment]
I notice the same sort of failures in #23545 - so I'd conclude its a new code bug in cmssw rather than this PR.

**Comment by smuzaffar** [comment]
+externals
ok, so I am merging this PR now.

**Pull request merged**